packages/js/src/classic-editor/replacement-variables.js

```diff
@@ -0,0 +1,339 @@
+import { select } from "@wordpress/data";
+import { __, sprintf } from "@wordpress/i18n";
+import { createDefaultReplacementVariableConfigurations, SEO_STORE_NAME } from "@yoast/seo-integration";
+import { flatten, get, map, once, pick, values } from "lodash";
+
+/**
+ * Gets the parent title from the select element.
+ *
+ * @param {HTMLElement} select The select input.
+ *
+ * @returns {string} The parent title.
+ */
+const getParentTitle = ( select ) => {
+	const title = get( select, `options.${ select?.selectedIndex }.text`, "" );
+
+	return title === __( "(no parent)", "wordpress-seo" ) ? "" : title;
+};
+
+/**
+ * Gets a set of replacement variable configurations, keyed by name for easy reference.
+ *
+ * These are the DRYed configurations, used in creating the sets per content type.
+ * When a replacement differs per content type, it might not belong here.
```

Comment:
They need to live inside a function because the translations have a race condition.
I did think about having a function for each replacement variable, but that seemed overkill.